website/static/js/pages/component-settings-page.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+
+var $ = require('jquery');
+var bootbox = require('bootbox');
+var Raven = require('raven-js');
+var $osf = require('js/osfHelpers.js');
+var { getConfirmationCode } = require('js/projectSettings.js');
+
+
+
+$(document).ready(function() {
+
+  //Project overview >> Component Widget >> Event/Click handler for componentQuickActions menu item: Delete.
+  $('.deleteComponent').each( function() {
+    $(this).off().on('click', function(e) {
+      var componentIndex = $(this).data('index');
+      var component = window.contextVars.nodes[componentIndex].node;
+
+      var {node_type, api_url, isPreprint, childExists} = component;
```

Comment:
Probably can just access via `component.api_url`, etc. I don't believe the `{var_a, var_B} = obj` pattern is used at all in the osf (though I don't know if there was a particular reason why).